dask/dataframe/io/parquet/utils.py

```diff
@@ -429,3 +430,18 @@ def _scrub(i, p):
         "/".join(basepath),
         out_list,
     )  # use '/'.join() instead of _join_path to be consistent with split('/')
+
+
+def unique_everseen(iterable):
+    """List unique elements, preserving order. Remember all elements ever seen.
+    FixMe: This has been pasted from https://github.com/dask/fastparquet/pull/471
+    and is not needed with fastparquet>0.3.2.
+
+    unique_everseen('AAAABBBCCDAABBB') --> A B C D
+    """
+
+    seen = set()
+    seen_add = seen.add
+    for element in itertools.filterfalse(seen.__contains__, iterable):
+        seen_add(element)
+        yield element
```

Comment:
Perhaps `toolz.unique` could do this?

Reply:
Now using `toolz.unique` (implementation is identical to `unique_everseen`), and `cytoolz.unique` if available - not that it's a bottleneck...

Reply:
Does this change need to be propagated back to `fastparquet`? Ideally the implementations wouldn't diverge.

Reply:
Fastparquet doesn't currently depend on `toolz`, hence the original small function - if we are going to require a `fastparquet` version, I would just remove the duplicate implementation here?

Reply:
I thought that a new enough fastparquet had been out for a few months. I think we may want to hold off on requiring something that new in case there are issues with the 0.3.3 release.

So right now my preference is to include this PR has is, and come back in a month or so to bump the required fastparquet version to 0.3.3.